server/controllers/users.js

```diff
@@ -185,11 +190,54 @@ const viewAllOffices = async (req, res) => {
   return res.status(codes.okay).json({ status: res.statusCode, data: offices });
 };
 
+const castVote = async (req, res) => {
+  // Fetch user ID
+  const { id: userId } = req.user;
+  if (!userId)
+    return res
+      .status(codes.unauthorized)
+      .json({ status: res.statusCode, error: messages.noToken });
```

Comment:
would mind creating a function for this? since you are repeating the same process many times